go/arrow/csv/reader.go

```diff
@@ -773,6 +778,18 @@ func (r *Reader) parseBinaryType(field array.Builder, str string) {
 	field.(*array.BinaryBuilder).Append(decodedVal)
 }
 
+func (r *Reader) parseExtension(field array.Builder, str string) {
+	if r.isNull(str) {
+		field.AppendNull()
+		return
+	}
+	dec := json.NewDecoder(strings.NewReader(`"` + str + `"`))
```

Comment:
Given this is a reader and it comes from the CSV it will always be a string, no?

Reply:
the raw value will be a string, but for example "3" would decode to the string "3" while the underlying storage or the custom unmarshalling may expect the number `3` instead. Essentially my objection here is the explicit addition of quotes and explicitly requiring JSON decoding. Also, if we're going to go this direction, we should at least use `strconv.Quote` instead :smile:

Reply:
I can go that direction, just needs some more guidance - so Im going to add an `AppendFromText` to the `Builder` interface. This will mean we will have to implement it in all types (which is fine). If yes, then Im going to open a new PR for that.

Reply:
I like the idea of `AppendFromText` because that could remove the need for the separate parse methods here. You'd just need to first check `r.isNull(str)`  before you could hand it off to the `AppendFromText` and have that return an `error` so it can error if it fails to parse the value. I like it. For the time being I think it's fine to leave the JSON style extension handling here for now and make that improvement as a separate PR.

Reply:
Can you file an issue to track this, referencing this PR? I'll merge this after the issue gets created.

Reply:
Done. Tacked here - https://github.com/apache/arrow/issues/34695